### ISSUE TITLE
False positive when using more than a symbol in schema.core/defn return value

### ISSUE THREAD

**Issue by ccann** [borkdude/clj-kondo#107]
e.g.

```clojure
(ns foo.bar.baz
  (:require [schema.core :as s :refer [Str]]))

(s/defn decode :- (s/cond-pre (s/pred map?) Str)
  [token & [unsign-opts second-try?]])
```
if you replace `(s/cond-pre (s/pred map?) Str)` with `Str` the warning goes away, but something like `(s/pred odd?)` also triggers it.

lint: `src/foo/bar/baz.clj:0:0: error: can't parse src/foo/bar/baz.clj, Don't know how to create ISeq from: clojure.lang.Symbol`

(this report presupposes you intend to support Schema)

**Comment by borkdude** [comment]
@ccann Thanks! I'll look into it.

**Comment by borkdude** [comment]
@ccann Can you also check if this issue is fixed with the binary provided in #106?

**Comment by ccann** [comment]
Sure is! Found a `redundant let` and a `redundant do`, to boot. (I didn't realize `do` in the body of a `try` was redundant, TIL)
